_posts/2024-05-04-vertexbuffer-docs-fix.md

```diff
@@ -0,0 +1,150 @@
+
+## Community Evaluation
+I worked with MonoGame for my bug fix. While I had a few issues with their maintainers not being the most communicative, they were managing some high-level bounties at the time and transferring their documentation over to an entirely new repo, so it was understandable that I wasn't exactly a high priority. 
+
+The guy who originally posted about this documentation issue way back in 2020 actually reacted to my desire to fix it! I was surprised he was even still paying attention to it at this point and it was nice to receive a positive reaction from a member of the community directly involved in the problem.
+
+## The Issue
+The issue I chose to solve is [Issue #7396 - [Doc] Incomplete VertexBuffer documentation](https://github.com/MonoGame/MonoGame/issues/7396). I picked a documentation issue mostly as I’ve been hearing complaints about the lack of clear documentation in MonoGame. I also personally dislike the lack of clarity in a lot of MonoGame documentation and how you mostly just have to figure out what to do entirely on your own. This particular issue was using magic numbers in an example, with no explanation as to how those values came about. While it's a useful guide for the one buffer they're displaying, that example is useless if you're working with any other buffer MonoGame provides.
+
+The issue has also been up since 2020, so either no one has bothered to fix it, or no one’s made an adequate change to the documentation that actually clarifies the SetData method. 
+The issue appeared to be something I could solve with a little bit of effort and would mostly rely on me figuring out how MonoGame's documentation system works. I have experience with handling (and screwing up) vertex buffers, so it was something well within my wheelhouse. The biggest task was actually finding what the docs were describing and working through what the code actually did.
+
+The same two maintainers I saw on my bug fix have also interacted with this issue organizationally. It’s gone from being a milestone for 3.8.1 to just an item on a To Do list for documentation. From the looks of it, despite MonoGame listing on their website that they swap maintainers monthly, these two will continue to be managing the repos for the time being.
+
+I commented on the issue thread and started the process of improving the lack of documentation.
+
+![Issue by georg-eckert-zeiss describing lack of detail in examples for Set Data in Vertex Buffer for MonoGame](../assets/2024-05-04-Allen/Issue#7396.png)
```

Comment:
# is not typically allowed in URLs and is probably why this image isnt working

Reply:
@h3allen not sure if you got a notification for the above so i figured a ping may be useful

Reply:
They showed up for me the second time (I think?)

Reply:
Haha nevermind